spec/controllers/v2/editions_controller_spec.rb

```diff
@@ -0,0 +1,122 @@
+require "rails_helper"
+
+RSpec.describe V2::EditionsController do
+  describe "index" do
+    before do
+      Timecop.freeze("2017-01-01 09:00:00") do
+        50.times do |index|
+          FactoryGirl.create(
+            :draft_edition,
+            id: index + 1,
+            document: FactoryGirl.create(:document, locale: "fr"),
+            base_path: "/content#{index + 1}",
+          )
+        end
+
+        50.times do |index|
+          FactoryGirl.create(
+            :live_edition,
+            id: index + 51,
+            document: FactoryGirl.create(:document),
+            base_path: "/content#{index + 51}",
+          )
+        end
+
+        50.times do |index|
+          FactoryGirl.create(
+            :unpublished_edition,
+            id: index + 101,
+            document: FactoryGirl.create(:document),
+            base_path: "/content#{index + 101}",
+            publishing_app: "test",
+          )
+        end
+      end
+    end
+
+    context "all editions" do
+      it "returns the correct list for the first page" do
+        get :index
+        expect(parsed_response["results"].count).to eq(100)
+        expect(parsed_response["links"]).to eq([
+          { "href" => "http://test.host/v2/editions?page=2017-01-01T09%3A00%3A00Z%2C100", "rel" => "next" },
```

Comment:
Could be good to have a urlencode function on an unencoded string so it's easier for the human brain to parse.